expertsystem/state/particle.py

```diff
@@ -258,6 +264,198 @@ def convert_to_dict(
 }
 
 
+class KinematicRepresentation:
+    def __init__(
+        self,
+        final_state: Optional[Union[Sequence[List[Any]], List[Any]]] = None,
+        initial_state: Optional[Union[Sequence[List[Any]], List[Any]]] = None,
+    ) -> None:
+        self.__initial_state: Optional[List[List[Any]]] = None
+        self.__final_state: Optional[List[List[Any]]] = None
+        if initial_state is not None:
+            self.__initial_state = self.__import(initial_state)
+        if final_state is not None:
+            self.__final_state = self.__import(final_state)
+
+    @property
+    def initial_state(self) -> Optional[List[List[Any]]]:
+        return self.__initial_state
+
+    @property
+    def final_state(self) -> Optional[List[List[Any]]]:
+        return self.__final_state
```

Comment:
is that really needed? To me only the equality and subset operators are important

Reply:
It is needed in the `__eq__` and subset (`__contains__`) operators when comparing with `other`
https://github.com/ComPWA/expertsystem/blob/9faf80e51c0835b34293c0c782d83fa377f4443a/expertsystem/state/particle.py#L334-L336

Reply:
But a `self.__initial_state` would suffice then

Reply:
Yes, but `other.initial_state` and `other.final_state` won't work

Reply:
interesting, somehow i was thinking in the friend scope of c++. I guess `_initial_state` would be the python equivalent. Well i leave it up to you

Reply:
Yes `_initial_state` would be possible. But then it's still accessible (you will only be warned by the linters). So I guess I'll leave it like this, now both members only have read access.